Scheduled tasks will not close the connection, causing losing connection to MySQL server during query

According to my observation, each scheduled task will have its own database connection, but it will not close the connection by itself after the task ends. When this number of connections reaches a certain number, the program can no longer create new connections, resulting in the inability to perform subsequent tasks.

![2020-05-04 11-45-51 的屏幕截图](https://user-images.githubusercontent.com/20673460/80934933-fe8ab680-8dfc-11ea-99ba-3aa2ca6d28a6.png)

I have just run into this issue as well with psql jobstore for django. 

`[ERROR]  Mysql closed the connection. Perform reconnect...`

I was able to replicate it on wsgi server by letting the backgroundscheduler start, confirming it had a connection entry in the psql server, then rebooting the psql server. The next job that runs throws the error above. All errors afterwards are thrown from jobstores.py trying to use the psql query backend.

`Traceback (most recent call last):
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django_apscheduler/jobstores.py", line 193, in __call__
    self._process_execution_event(event)
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django_apscheduler/jobstores.py", line 212, in _process_execution_event
    job = DjangoJob.objects.get(name=event.job_id)
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/models/manager.py", line 82, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/models/query.py", line 393, in get
    num = len(clone)
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/models/query.py", line 250, in __len__
    self._fetch_all()
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/models/query.py", line 1183, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/models/query.py", line 54, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/models/sql/compiler.py", line 1059, in execute_sql
    cursor = self.connection.cursor()
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/backends/base/base.py", line 255, in cursor
    return self._cursor()
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/backends/base/base.py", line 234, in _cursor
    return self._prepare_cursor(self.create_cursor(name))
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/backends/base/base.py", line 234, in _cursor
    return self._prepare_cursor(self.create_cursor(name))
  File "/home/site/wwwroot/antenv/lib/python3.7/site-packages/django/db/backends/postgresql/base.py", line 218, in create_cursor
    cursor = self.connection.cursor()
django.db.utils.InterfaceError: connection already closed`
[dj_aps_bug.txt](https://github.com/jarekwg/django-apscheduler/files/4669347/dj_aps_bug.txt)

--- a/django_apscheduler/jobstores.py
+++ b/django_apscheduler/jobstores.py
@@ -9,6 +9,7 @@ from apscheduler.jobstores.base import BaseJobStore, ConflictingIdError, JobLook
 from apscheduler.schedulers.base import BaseScheduler
 
 from django.core.exceptions import ObjectDoesNotExist
+from django import db
 from django.db import connections
 from django.db.utils import OperationalError, ProgrammingError
 
@@ -17,6 +18,7 @@ from django_apscheduler.result_storage import DjangoResultStorage
 from django_apscheduler.util import deserialize_dt, serialize_dt
 
 LOGGER = logging.getLogger("django_apscheduler")
+
 
 def ignore_database_error(on_error_value=None):
 
@@ -35,6 +37,8 @@ def ignore_database_error(on_error_value=None):
                     stacklevel=3
                 )
                 return on_error_value
+            finally:
+                db.connections.close_all()
         return inner
     return dec
 
@@ -194,6 +198,7 @@ class _EventManager(object):
         except Exception as e:
             self.LOGGER.exception(str(e))
 
+    @ignore_database_error()
     def _process_submission_event(self, event):
         # type: (JobSubmissionEvent)->None
 
@@ -205,6 +210,7 @@ class _EventManager(object):
 
         self.storage.get_or_create_job_execution(job, event)
 
+    @ignore_database_error()
     def _process_execution_event(self, event):
         # type: (JobExecutionEvent)->None
 

